lib/YaST/Firstboot/FirstbootController.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Controller for firstboot client
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package YaST::Firstboot::FirstbootController;
+use strict;
+use warnings;
+use YuiRestClient;
+use YaST::Firstboot::GenericClient;
+use YaST::Firstboot::LanClient;
+use YaST::NetworkSettings::OverviewTab;
+use YaST::NetworkSettings::NetworkCardSetup::AddressTab;
+use YaST::NetworkSettings::NetworkCardSetup::GeneralTab;
+use YaST::NetworkSettings::NetworkCardSetup::VLANAddressTab;
+use testapi;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {}, $class;
+    return $self->init($args);
+}
+
+sub init {
+    my ($self, $args) = @_;
+    $self->{GenericClient} = YaST::Firstboot::GenericClient->new({app => YuiRestClient::get_app()});
+    $self->{LanClient} = YaST::Firstboot::LanClient->new({app => YuiRestClient::get_app()});
+    return $self;
+}
+
+sub generic_smoketest {
```

Comment:
names like 'smoketest' 'test' or 'assert' are clear indicator of some smelly code, not the name is not right, it is that framework is a testing framework and it doesn't make sense in general those there, this words we should keep it for the test modules if corresponding. 'client' is also part of the YaST implementation, so for us are just screens.

In other words, we can just call it by the name of what they are doing even if it is a simple thing.
Anyway I think we agreed to have only the setup, so if you can check something on the first screen/client that's it, (not even try to go to second with lan tabs) rest api works, job done. We will design later how to do it properly and not repeat the code that we have in installation.

Reply:
I'm not sure I get you completely here. It looks like there are two things. one is the name of that function, which I could rename for example check_and_skip_page (?) and the other is the usage of "client" rather than "page". Is that what you mean ?

Reply:
Depending of what you want to check, which I don't know, something like 'get_text_<of_something>' in the welcome page and not even bother to do anything else with the REST API, click next with testapi. So there is no need for those Network objects. This code should be extended so the idea is to do really the minimum  here and leave a code with a simple comment that anyone can understand even without syncing with us.

Reply:
> Depending of what you want to check, which I don't know

replace the assert_screens with some other way of asserting the page. This is why we talked about implementing the posibility to probe the debug_labels, as they are the only element common to each page.

Reply:
> So there is no need for those Network objects. This code should be extended so the idea is to do really the minimum here and leave a code with a simple comment that anyone can understand even without syncing with us.

Are you saying we don't want to use libyui ? I'm confused.

Reply:
I think that code would be reworked so my only point is to check something and do the minimum, not even visit two screens/clients, is not important if you use next from REST API or not.